src/ensembl/src/content/app/species/state/general/speciesGeneralHelper.ts

```diff
@@ -0,0 +1,477 @@
+  [Stats.AVERAGE_SEQUENCE_LENGTH]: { label: 'Average sequence length' },
+  [Stats.AVERAGE_CDS_LENGTH]: { label: 'Average CDS length' },
+  [Stats.TOTAL_TRANSCRIPTS]: { label: 'Total transcripts' },
+  [Stats.CODING_TRANSCRIPTS]: { label: 'Coding transcripts' },
+  [Stats.TRANSCRIPTS_PER_GENE]: { label: 'Transcripts per gene' },
+  [Stats.CODING_TRANSCRIPTS_PER_GENE]: { label: 'Coding transcripts per gene' },
+  [Stats.TOTAL_EXONS]: { label: 'Total exons' },
+  [Stats.TOTAL_CODING_EXONS]: { label: 'Total coding exons' },
+  [Stats.AVERAGE_EXON_LENGTH]: { label: 'Average exon length' },
+  [Stats.AVERAGE_CODING_EXON_LENGTH]: { label: 'Average coding exon length' },
+  [Stats.AVERAGE_EXONS_PER_TRANSCRIPT]: {
+    label: 'Average exons per transcript'
+  },
+  [Stats.AVERAGE_CODING_EXONS_PER_CODING_TRANSCRIPT]: {
+    label: 'Average coding exons per coding transcript'
+  },
+  [Stats.TOTAL_INTRONS]: { label: 'Total introns' },
+  [Stats.AVERAGE_INTRON_LENGTH]: { label: 'Average intron length' },
+  [Stats.TOTAL_GAP_LENGTH]: { label: 'Total gap length' },
+  [Stats.SPANNED_GAP]: { label: 'Spanned gaps' },
+  [Stats.TOPLEVEL_SEQUENCES]: { label: 'Toplevel sequences' },
+  [Stats.COMPONENT_SEQUENCES]: { label: 'Component sequences' },
+  [Stats.ASSEMBLY_NAME]: { label: 'Assembly name' },
+  [Stats.ASSEMBLY_DATE]: { label: 'Assembly date' },
+  [Stats.ASSEMBLY_ACCESSION]: { label: 'Assembly accession' },
+  [Stats.CONTIG_N50]: { label: 'Contig N50' },
+  [Stats.TAXONOMY_ID]: { label: 'Taxonomy id' },
+  [Stats.STRAIN]: { label: 'Breed/Cultivar/Strain' },
+  [Stats.SEX]: { label: 'Sex' },
+  [Stats.SCIENTIFIC_NAME]: { label: 'Scientific name' },
+  [Stats.PSEUDOGENES]: { label: 'Pseudogenes' }
+};
+
+type StatsGroup = {
+  title: string;
+  stats: [IndividualStat[]];
+};
+
+export type StatsSection = {
+  section: SpeciesStatsSection;
+  exampleLinks?: Partial<urlObj>;
+  primaryStats?: IndividualStat;
+  secondaryStats?: IndividualStat;
+  groups: StatsGroup[];
+};
+
+type BuildStatProps = Partial<IndividualStat> & {
+  primaryValue: string | number;
+  primaryKey: Stats;
+  secondaryKey?: Stats;
+};
+
+const buildIndividualStat = (
+  props: BuildStatProps
+): IndividualStat | undefined => {
+  let primaryValue = props.primaryValue;
+
+  const { primaryValuePostfix, label, primaryUnit } = statsFormattingOptions[
+    props.primaryKey
+  ];
+
+  if (typeof primaryValue === 'number') {
+    primaryValue =
+      getCommaSeparatedNumber(primaryValue) +
+      (primaryValuePostfix ? primaryValuePostfix : '');
+  }
+
+  return {
+    label: label || props.primaryKey,
+    primaryValue,
+    primaryUnit: primaryUnit
+  };
+};
+
+type BuildHeaderStatProps = {
+  primaryValue: string | number;
+  primaryKey: Stats;
+};
+
+const buildHeaderStat = (
+  props: BuildHeaderStatProps
+): IndividualStat | undefined => {
+  let primaryValue = props.primaryValue;
+
+  const { primaryValuePostfix, headerUnit } = statsFormattingOptions[
```

Comment:
if `StatsFormattingOptions` type really has a field `{primaryValuePostfix?: string;}`, then you can use the default value for primaryValuePostfix when destructuring (`const { primaryValuePostfix = '' , ... } = ...`). Then you wouldn't need to apply a ternary operator to check that primaryValuePostfix exists below.

Reply:
That's done